src/ray/raylet/scheduling_resources.cc

```diff
@@ -241,9 +307,18 @@ ResourceIds ResourceIds::Acquire(double resource_quantity) {
 void ResourceIds::Release(const ResourceIds &resource_ids) {
   auto const &whole_ids_to_return = resource_ids.WholeIds();
 
-  // Return the whole IDs.
-  whole_ids_.insert(whole_ids_.end(), whole_ids_to_return.begin(),
-                    whole_ids_to_return.end());
+  int64_t return_resource_count = whole_ids_to_return.size();
+  if (return_resource_count > decrement_backlog_) {
```

Comment:
We're only clearing the `decrement_backlog_` when we return whole IDs. So if each task only uses fractional resources, then we will never clear the `decrement_backlog_`, right?